### PR TITLE
Stops multiple galley image files with the same name from being loaded in the edit galley interface.

### LINKED ISSUE #3979: Multiple galley images with the same name results in an error
**Describe the bug**
If an editor/typesetter/production manager uploads two images with the same name this results in the image is not displayed. 

**Janeway version**
1.5.1-RC-5

**Pontential Solutions**
- [x] For legacy we should add a handle for this in the `identifier_figure` view
- [x] When an image is uploaded that already exists, warn the user and recommend they use the replace file option
- [x] If the replace file option isn't present on this interface, add it

### PR DESCRIPTION
**Pull request by ajrbyers**
This closes #3979 

The implementation includes two prongs:

`save_galley_image` now takes `check_for_existing_images` which defaults to `False`. When set to `True` the function will check if the image exists and if its found return `None`. Existing calls are protected.

In addition, I've updated the Typesetting Plugin view to use the new kwarg: https://github.com/BirkbeckCTP/typesetting/pull/204

### CODE REVIEW

**Comment by ajrbyers** [comment]
Ignore this for now, I need to alter the comment, as I swapped from .first() to .latest()

File: src/production/logic.py [hunk 1]
```diff
@@ -174,22 +174,39 @@ def replace_galley_file(article, request, galley, uploaded_file):
 
 
 def save_galley_image(galley, request, uploaded_file, label="Galley Image", fixed=False):
-
+    filename = uploaded_file.name
     if fixed:
```
Comment (mauromsl) [inline src/production/logic.py:178]: Shouldn't there be a kwarg here called `check_for_existing images` as per https://github.com/BirkbeckCTP/typesetting/pull/204 ?

**Review by mauromsl — CHANGES_REQUESTED** [review]
I think you forgot to add support for the lastest feature

File: src/production/logic.py (hunk 1 above)
Comment (ajrbyers) [reply src/production/logic.py]: I had the commit but hadn't pushed it :D

**Comment by joemull** [comment]
@ajrbyers I will review this once that commit is added because I want to test out the interface.

**Comment by joemull** [comment]
Or is it now ready?

**Comment by ajrbyers** [comment]
Hey @joemull, it is ready now the commit is here: https://github.com/BirkbeckCTP/janeway/pull/3981/commits/726dd480b4c449313f9608243a9b7f63803d8eb0

File: src/production/logic.py [hunk 2]
```diff
@@ -173,23 +173,46 @@ def replace_galley_file(article, request, galley, uploaded_file):
...
+    if check_for_existing_images and galley.images.filter(
+        original_filename=filename,
+    ).exists():
+        messages.add_message(
+            request,
+            messages.WARNING,
+            f'An image called {filename} already exists. Use the replace '
+            f'file function to upload a new version.',
+        )
+        return
```
Comment (mauromsl) [inline src/production/logic.py:196]: I think also changing the default behaviour to override the matched figure by filename and galley ID with a new version is a bit tidier (As opposed to having duplicate files with idential names).
I understand that the patch to `article_figure` does resolve the reported problem, however we might see the problem reoccur on other views with similar logic.

**Review by mauromsl — COMMENTED** [review]
It fixes the reported issue, but made a suggestion on implementation that I think will lead to a cleaner filesystem, especially for cases where images are bulk-loaded

File: src/production/logic.py [hunk 3]
```diff
@@ -173,23 +173,46 @@ def replace_galley_file(article, request, galley, uploaded_file):
...
         expected_mime = files.guess_mime(filename)
 
         if not uploaded_file_mime == expected_mime:
-            messages.add_message(request, messages.WARNING, 'The file you uploaded does not match the mime of the '
-                                                            'file expected.')
+            messages.add_message(
+                request,
+                messages.WARNING,
+                'The file you uploaded does not match the mime of the '
+                'file expected.',
```
Comment (joemull) [inline src/production/logic.py:207]: This is a good opportunity to make this text a little more user-friendly. Most people don't know what MIME types are, and we also aren't showing users potentially helpful information (e.g. what the expected type is). Maybe we could do something like `f'The file you uploaded does not have the expected type: { expected_mime }.'`

**Review by joemull — CHANGES_REQUESTED** [review]
Tested it out and I like the behavior. Just one request to improve some help text.

File: src/production/logic.py (hunk 3 above)
Comment (ajrbyers) [reply src/production/logic.py:207]: Thanks Joe. Will update.

**Comment by joemull** [comment]
This has been superseded by #4032.

**Pull request closed without merging**